src/kvtree_util.c

```diff
@@ -225,3 +225,149 @@ int kvtree_util_get_ptr(const kvtree* hash, const char* key, void** val)
 
   return rc;
 }
+
+/*
+ * Read the contents of a summary file (rank2file) and all of it's subfiles
+ * (rank2file.x.y) into a single big kvtree.  This is similiar to
+ * kvtree_read_scatter() but non-MPI.
+ *
+ * prefix:  The directory path containing the rank2file* files.
+ * data:    A new, empty, kvtree that gets filled in by this function.
+ *
+ * Returns KVTREE_SUCCESS on success, non-KVTREE_SUCCESS otherwise.
+ */
+int kvtree_read_scatter_single(const char* prefix, kvtree* data)
+{
+  char path[PATH_MAX];
+  kvtree* subfile_kvtree;
+  int rc;
+  kvtree_elem* elem;
+  kvtree* final_rank_kvtree;
+
+  /* Read in the contents of the main 'rank2file' tree. */
+  snprintf(path, sizeof(path), "%s/rank2file", prefix);
```

Comment:
Let's drop the "rank2file" part here and all other references to it.  That is a prefix name passed in from the user, in this case it's coming from SCR.  The user is free to pass in any prefix string they want for this part.  I think it should be find to just use ``snprintf(path, sizeof(path), "%s", prefix);``